src/jaegertracing/TracerFactory.cpp

```diff
@@ -38,19 +38,16 @@ TracerFactory::MakeTracer(const char* configuration,
             opentracing::configuration_parse_error);
     }
 
-    const auto serviceNameNode = yaml["service_name"];
-    if (!serviceNameNode) {
+    auto tracerConfig = jaegertracing::Config::parse(yaml);
+
+    tracerConfig.fromEnv();
+
+    if (tracerConfig.serviceName().empty()) {
         errorMessage = "`service_name` not provided";
         return opentracing::make_unexpected(
             opentracing::invalid_configuration_error);
     }
-    if (!serviceNameNode.IsScalar()) {
```

Comment:
not sure what this was testing, why is it no longer needed?

Reply:
This was manually verifying the integrity of the yaml, which should be done by the parsing itself